### PR TITLE
fix: Add keyboard support to CardHorizontalBarChart and InfoTooltip

### LINKED ISSUE #4133: enhancement: a11y rule "mouse-events-have-key-events" fails with a warning.
### Describe the bug

[jsx-a11y eslint plugin](https://www.npmjs.com/package/eslint-plugin-jsx-a11y) was recently added to the project and this rule -
jsx-a11y/mouse-events-have-key-events - fails with a warning.

This issue offers more context: https://github.com/open-sauced/app/issues/2264

### Steps to reproduce

The following components and files are causing the rule to fail:

1. components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx
2. components/shared/InfoTooltip.tsx

### PR DESCRIPTION
**Pull request by Firdous2307**
## Description

This PR fixes accessibility issues in the CardHorizontalBarChart and InfoTooltip components. It addresses the `jsx-a11y/mouse-events-have-key-events` warnings and improves keyboard navigation support.

## Related Tickets & Documents

Fixes #4133

## Mobile & Desktop Screenshots/Recordings

N/A (No visual changes, only accessibility improvements)

## Steps to QA

1. Navigate to the CardHorizontalBarChart component using a keyboard
2. Verify that you can focus on individual language bars using the Tab key
3. Press Enter or Space on a focused bar to change the description text
4. Navigate to a page containing the InfoTooltip component
5. Verify that you can open and close the tooltip using both mouse and keyboard interactions

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [ ] Tier 4

## [optional] What gif best describes this PR or how it makes you feel?

Accessibility

### CODE REVIEW

File: components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx [hunk 1]
```diff
@@ -55,6 +61,11 @@ const CardHorizontalBarChart = ({ languageList, withDescription }: CardHorizonta
               <div
                 key={index}
                 onMouseOver={() => handleChangeDescriptText(languageName)}
+                onFocus={() => handleChangeDescriptText(languageName)}
+                onKeyDown={(e) => handleKeyDown(e, languageName)}
+                tabIndex={0}
+                role="button"
+                aria-label={`${languageName}: ${percentageUsed}%`}
```
Comment (nickytonline) [inline components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx:68]: tabIndex and role props aren't necessary. Use a button instead. Also include button text, i.e. `<span className="sr-only">{`languageName ${percentageUsed}%`}</span>`

```jsx
              <button
                key={index}
                onMouseOver={() => handleChangeDescriptText(languageName)}
                onFocus={() => handleChangeDescriptText(languageName)}
                onKeyDown={(e) => handleKeyDown(e, languageName)}
                aria-label={`${languageName}: ${percentageUsed}%`}
                className="h-2 transition-all duration-500 ease-in-out"
                style={{
                  width: `${percentageUsed < 20 ? (percentageUsed / percentage) * 100 : percentageUsed}%`,
                  backgroundColor: languageToColor[languageName]
                    ? (languageToColor[languageName].color as string)
                    : NOTSUPPORTED,
                }}
              >
                <span className="sr-only">{`languageName ${percentageUsed}%`}</span>
              </button>
```

**Review by nickytonline — CHANGES_REQUESTED** [review]

File: components/shared/InfoTooltip.tsx [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
+import { useState, useCallback } from "react";
 import { TooltipTrigger, TooltipPortal, Tooltip, TooltipContent, TooltipArrow } from "@radix-ui/react-tooltip";
 import { HiOutlineInformationCircle } from "react-icons/hi";
 
 export default function InfoTooltip({ information, icon }: { information: string; icon?: React.ReactNode }) {
   const [open, setOpen] = useState(false);
 
+  const handleOpenChange = useCallback((isOpen: boolean) => {
+    setOpen(isOpen);
+  }, []);
```
[the remark this answers is not in the document]
Comment (Firdous2307) [reply components/shared/InfoTooltip.tsx:10]: Thanks for the feedback @nickytonline

File: components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx (hunk 1 above)
Comment (Firdous2307) [reply components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx]: @nickytonline I have implemented the changes you suggested.

File: components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx [hunk 3]
```diff
@@ -52,17 +58,22 @@ const CardHorizontalBarChart = ({ languageList, withDescription }: CardHorizonta
         {sortedLangArray.map(({ languageName, percentageUsed }, index) => {
           return (
             index < 5 && (
-              <div
+              <button
                 key={index}
                 onMouseOver={() => handleChangeDescriptText(languageName)}
+                onFocus={() => handleChangeDescriptText(languageName)}
+                onKeyDown={(e) => handleKeyDown(e, languageName)}
+                aria-label={`${languageName}: ${percentageUsed}%`}
```
Comment (nickytonline) [inline components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx:66]: Now that we're using `<span className="sr-only">{`languageName ${percentageUsed}%`}</span>` this is no longer needed. I missed that in my last review.

```suggestion
```

File: components/shared/InfoTooltip.tsx [hunk 4]
```diff
@@ -1,14 +1,18 @@
-import { useState } from "react";
+import { useState, useCallback } from "react";
```
Comment (nickytonline) [inline components/shared/InfoTooltip.tsx:1]: `useCallback` is no longer being used, so this import can be removed.

```suggestion
import { useState } from "react";
```

**Review by nickytonline — CHANGES_REQUESTED** [review]
This is looking great @Firdous2307. Just a couple other things to remove and this is good to go. Thanks for your patience!

**Review by nickytonline — CHANGES_REQUESTED** [review]

**Comment by Firdous2307** [comment]
> Hi @Firdous2307, just seeing if you're blocked on anything. No rush on this, just checking in.

@nickytonline Everything is good, I just took a little break, thanks for checking in.

File: components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx [hunk 5]
```diff
@@ -35,6 +35,12 @@ const CardHorizontalBarChart = ({ languageList, withDescription }: CardHorizonta
     setDescriptText(descriptText);
   };
 
+  const handleKeyDown = (event: React.KeyboardEvent, languageName: string) => {
```
[the remark this answers is not in the document]
Comment (CorinaMurg) [reply components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx:38]: @nickytonline @Firdous2307 
What will this `<span>` do? Remove the visible names and makes them available to screen readers only?

[the remark this answers is not in the document]
Comment (Firdous2307) [reply components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx]: All done @nickytonline

[the remark this answers is not in the document]
Comment (Firdous2307) [reply components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx]: > @nickytonline @Firdous2307 What will this `<span>` do? Remove the visible names and makes them available to screen readers only?

@CorinaMurg Yes, exactly!

[the remark this answers is not in the document]
Comment (nickytonline) [reply components/molecules/CardHorizontalBarChart/card-horizontal-bar-chart.tsx]: In most of the cases @CorinaMurg, they didn't have a label to begin with so this is already an improvement. Longer term it would be good to have the label visible so it's not just for screen readers, but I'd suggest a separate issue for that as it probably affects the current design a bit.

File: .github/workflows/issue.yml [hunk 6]
```diff
@@ -12,8 +12,8 @@ permissions:
 jobs:
   comment:
     permissions:
-      issues: write  # for peter-evans/create-or-update-comment to create or update comment
-      pull-requests: write  # for peter-evans/create-or-update-comment to create or update comment
+      issues: write # for peter-evans/create-or-update-comment to create or update comment
+      pull-requests: write # for peter-evans/create-or-update-comment to create or update comment
```
Comment (nickytonline) [inline .github/workflows/issue.yml:16]: These changes aren't related to the fix. You can undo them by running `git checkout beta -- .github/workflows/issue.yml` and committing the changes to your branch and pushing them.

File: .gitpod.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# This configuration file was automatically generated by Gitpod.
```
Comment (nickytonline) [inline .gitpod.yml:1]: Refactor (blocking): Remove the .gitpod.yml file. The project doesn't use gitpod and this change is not related to the fix.

**Review by nickytonline — CHANGES_REQUESTED** [review]

**Comment by Firdous2307** [comment]
@nickytonline I mistakenly closed the main repository, thinking the changes were already merged. Apologies for any inconvenience caused.
